### ISSUE TITLE
Improve 'raise' to introduce let bindings for newly out-of-scope vars

### ISSUE THREAD

**Issue by brprice** [hackworthltd/primer#56]
This is a placeholder for now, I will come back and fill this in later.

If we want to do this generically for copy/paste, we may want to use letType to deal with typevars. This would need us to fix hackworthltd/primer#5.
However, we may not want to do this, for consistency with the type level copy/paste which has no lets. ~~It is also a bit pointless as it wouldn't be exposed to the user of `raise`, as you cannot hit this situation with that tool.~~ (Well, you could do `? : ∀a . List a`, and try to lift the `List a` to get `? : List ?`. We could instead give `lettype a = ? in ? : List a`, by introducing a `lettype` outside the annotation, but I suspect this is more confusing than it is worth!) **Edit:** scrap that, you could have `Λa . (? : List a)`, and raise this to get `? : List ?`.

We will want to consider how to make this non-annoying when lifting the body of a lambda etc (or worse, a let!). Going from `(λx.t)` to `let x = ? in t` doesn't seem all that useful for `raise`!

## Description

This is a feature request.

<*Include a brief description here.*>

## Dependencies

<*Link to other GitHub Issues or PRs that must be completed before work can begin on this feature.*>

## Spec

<*Describe what the feature will do, from a high-level perspective.*>

## Implementation details

<*If there are any particularly tricky implementation bits that are worth discussing or you haven't quite figured out yet, describe the details here.*>

## Not in spec

<*Describe any features that your implementation is explicitly avoiding, that a reasonable person might think should be in spec. For example, if you're adding a new action that operates on variables, but it only works with term variables and not type variables, you might want to mention it here so that the scope of the feature is clear.*>

## Discussion

<*If there's a GitHub Discussions topic where this feature has been/is being discussed, link it here.*>

## Future work

<*Describe ideas or additional features that might be useful once this feature has been implemented. This is a good place to link to other GitHub Issues or PRs that track this future work.*>
